holoviews/plotting/bokeh/chart.py

```diff
@@ -906,25 +903,39 @@ def get_data(self, element, ranges, style):
         hover = 'hover' in self.handles
 
         # Group by stack or group dim if necessary
+        xdiff = None
+        xvals = element.dimension_values(xdim)
         if group_dim is None:
             grouped = {0: element}
+            is_dt = isdatetime(xvals)
+            if is_dt or xvals.dtype.kind != 'O':
+                xdiff = np.diff(xvals)
+                if len(np.unique(xdiff)) == 1 and xdiff[0] == 0:
+                    xdiff = 1
+                if is_dt:
+                    width = (xdiff.astype('timedelta64[ms]').astype(np.int64) * width)
+                    width = np.min(np.abs(width))
+                else:
+                    width = width / xdiff
+                    width = 1 - np.min(np.abs(width))
```

Comment:
Guess I'm still a little confused why this subtraction is needed? Won't this be negative in many cases?

Reply:
Also differs with the matplotlib implementation below.

Reply:
Yeah I'm not sure why there was a `1 -` Changed it.